src/plugins/CoreTests/src/AnnotationTableObjectTest.cpp

```diff
@@ -885,6 +885,101 @@ Task::ReportResult GTest_CheckAnnotationsNamesInTwoObjects::report() {
+                             .arg(obj->getGObjectName(), document->getName()));
+                return {};
+            }
+        return tables;
+    };
+
+    QString docName = doc->getName(),
+            docName2 = doc2->getName();
+    QList<AnnotationTableObject *> tables = getAnnotationTables(doc),
+                                   tables2 = getAnnotationTables(doc2);  // The order of the tables matters, the name
+                                                                         // and other characteristics do not matter.
+    if (hasError())
+        return ReportResult_Finished;
+    if (tables.size() != tables2.size() || tables.empty()) {
+        setError(QString("The first document '%1' has %2 annotation tables, the second '%3' has %4")
+                     .arg(docName)
+                     .arg(tables.size())
+                     .arg(docName2)
+                     .arg(tables2.size()));
+        return ReportResult_Finished;
+    }
+
+    for (int i = 0; i < tables.size(); i++) {
+        auto annotToString = [](Annotation *a) {
+            return a->getName() + ' ' + U2FeatureTypes::getVisualName(a->getType()) + ' ' +
+                   a->getGroup()->getGroupPath() + ' ' + buildSortedLocationString(a);
+        };
+        auto getAnnotationsNotComments = [](AnnotationTableObject *table) {
+            QList<Annotation *> anns = table->getAnnotations();
+            anns.erase(std::remove_if(anns.begin(),
+                                      anns.end(),
+                                      [](Annotation *a) { return a->getType() == U2FeatureType::Comment; }),
+                       anns.end());
+            return anns;
+        };
+        AnnotationTableObject *table = tables[i],
+                              *table2 = tables2[i];
+        QString tableName = table->getGObjectName(),
+                tableName2 = table2->getGObjectName();
+        QList<Annotation *> anns = getAnnotationsNotComments(table),
+                            anns2 = getAnnotationsNotComments(table2);  // Annotation order doesn't matter.
+        if (anns.size() != anns2.size()) {
+            setError(QString("Table '%1' ('%2') has %3 annotations, table '%4' ('%5') has %6")
+                         .arg(tableName, docName)
+                         .arg(anns.size())
+                         .arg(tableName2, docName2)
+                         .arg(anns2.size()));
+            return ReportResult_Finished;
+        }
+        for (Annotation *a : qAsConst(anns)) {
+            auto foundAnn = std::find_if(anns2.begin(), anns2.end(), [a](Annotation *a2) {
+                return *a->getData() == *(a2)->getData() &&
+                       a->getGroup()->getGroupPath() == a2->getGroup()->getGroupPath();  // Case sensitive.
+                                                                         // AnnotationData::operator== may contain bugs.
```

Comment:
Lets document/create this bug and point it here. Looks like something important

Reply:
If there are identical qualifiers in the annotation, then equality will not be a symmetric operation: {q1, q1} == {q1, q2} will return true, {q1, q2} == {q1, q1} will return false